pennylane/labs/zxopt/basic_optimization.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Optimization pass ``basic_optimization`` from pyzx using ZX calculus."""
+
+import pyzx as zx
+
+import pennylane as qml
+
+from .zx_conversion import _tape2pyzx
+
+
+def basic_optimization(tape):
+    r"""
+    Apply `zx.basic_optimization <https://pyzx.readthedocs.io/en/latest/api.html#pyzx.optimize.basic_optimization>`__ to a PennyLane `phase polynomial <https://pennylane.ai/compilation/phase-polynomial-intermediate-representation>`__ circuit with :class:`~Hadamard` gates.
+    This step can help improve phase polynomial based optimization schemes like :func:`~todd` or :func:`~full_optimize` by moving :class:`~Hadamard` to create big and few phase polynomial blocks.
+
+    Args:
+        tape (qml.tape.QuantumScript): Input PennyLane circuit.
+
+    Returns:
+        qml.tape.QuantumScript: T gate optimized PennyLane circuit.
+
+    .. seealso:: :func:`~full_reduce` (arbitrary circuits), :func:`~full_optimize` (`(Clifford + T) <https://pennylane.ai/compilation/clifford-t-gate-set>`__ circuits)
+
+    **Example**
+
+    This pass tried to push :class:`~Hadamard` gates as far as possible to the side to allow better phase polynomial optimization via, e.g., :func:`~todd`.
```

Comment:
To either side or a particular one? :) 
```suggestion
    This pass tries to push :class:`~Hadamard` gates as far as possible to the side to allow better phase polynomial optimization via, e.g., :func:`~todd`.
```